### PR TITLE
Feature/em 818 create new bundle details

### PR DESCRIPTION
**Pull request by velimd**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

Notes:
*created the bundle creation page for the user to give details about the bundle
*included validation (front end) with gds styles
*included the panel component

### CODE REVIEW

**Review by linusnorton — APPROVED** [review]

File: src/app/bundles/bundle-details/bundle-details.component.html [hunk 1]
```diff
@@ -1,3 +1,28 @@
-<p>
-  bundle-details works!
-</p>
+<app-panel [title]="'Bundle Details'" [description]="'Name bundle and provide description'"></app-panel>
+<div class="govuk-width-container">
+  <div id="bundle-details-container">
+    <form>
+      <div [className]="bundleName.touched && !bundleName.valid ? 'govuk-form-group govuk-form-group--error' : 'govuk-form-group'">
```
Comment (benouaer) [inline src/app/bundles/bundle-details/bundle-details.component.html]: I would rather move this logic out of here into a function/property that you can reference here

File: src/app/bundles/bundle-details/bundle-details.component.html [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
+        <div id="bundle-name">
+          <label class="govuk-label" for="name">
+            Bundle name <span>&#42;</span>
+          </label>
+          <span id="first-name-error" class="govuk-error-message" *ngIf="bundleName.touched && !bundleName.valid">
+            Enter the name for the bundle
+          </span>
+          <input [className]="bundleName.touched && !bundleName.valid ? 'govuk-input govuk-input--error' : 'govuk-input'" id="name" type="text" required [(ngModel)]="name" name="name" #bundleName="ngModel">
```
Comment (benouaer) [inline src/app/bundles/bundle-details/bundle-details.component.html]: same comment as above about externalising the class expression

File: src/app/bundles/bundle-details/bundle-details.component.ts [hunk 3]
```diff
@@ -7,9 +7,16 @@ import { Component, OnInit } from '@angular/core';
...
 
+  name: string;
+  description: string;
+
   constructor() { }
 
   ngOnInit() {
   }
 
+  saveBundleDetails() {
```
Comment (benouaer) [inline src/app/bundles/bundle-details/bundle-details.component.ts:19]: we might need to consider at least creating a dummy service that we invoke here. but might something for later

File: src/app/bundles/bundle-details/bundle-details.component.html [hunk 4]
```diff
@@ -1,3 +1,34 @@
...
+            Bundle name
+          </label>
+          <span id="first-name-error" class="govuk-error-message" *ngIf="bundleName.touched && !bundleName.valid">
+            Enter the name for the bundle
+          </span>
+          <input [ngClass]="bundleName.touched && !bundleName.valid ? 'govuk-input govuk-input--error' : 'govuk-input'"
+                 id="name"
+                 type="text"
+                 required
+                 [ngModel]="name"
```
Comment (linusnorton) [inline src/app/bundles/bundle-details/bundle-details.component.html]: How come this doesn't need to be two way binding? ie `[(ngModel)]`

File: src/app/bundles/bundle-details/bundle-details.component.html [hunk 5]
```diff
@@ -1,3 +1,34 @@
...
+        <div id="bundle-description">
+          <label class="govuk-label" for="description">
+            Bundle description
+          </label>
+          <textarea class="govuk-textarea" id="description" name="description" rows="5" aria-describedby="more-detail-hint" [(ngModel)]="description"></textarea>
+        </div>
+      </div>
+    </form>
+  </div>
+  <a routerLink="/bundle/import/:id" class="govuk-button" id="next" (click)="saveBundleDetails()">Save</a>
```
Comment (linusnorton) [inline src/app/bundles/bundle-details/bundle-details.component.html]: How does `:id` get populated here?

**Pull request merged**